eureka-client/src/main/java/com/netflix/discovery/DiscoveryClient.java

```diff
@@ -1381,15 +1382,27 @@ void refreshInstanceInfo() {
         applicationInfoManager.refreshLeaseInfoIfRequired();
 
         InstanceStatus status;
-        try {
-            status = getHealthCheckHandler().getStatus(instanceInfo.getStatus());
-        } catch (Exception e) {
-            logger.warn("Exception from healthcheckHandler.getStatus, setting status to DOWN", e);
-            status = InstanceStatus.DOWN;
-        }
+        ReentrantLock instanceStatusUpdateLock = new ReentrantLock();
```

Comment:
Should probably be a field if I understand correctly.

Reply:
Right. No idea why I put it this way.

Reply:
Thanks for the catch.